lib/libzfs/libzfs_dataset.c

```diff
@@ -2007,6 +2007,7 @@ zfs_prop_inherit(zfs_handle_t *zhp, const char *propname, boolean_t received)
 		goto error;
 
 	if ((ret = zfs_ioctl(zhp->zfs_hdl, ZFS_IOC_INHERIT_PROP, &zc)) != 0) {
+		changelist_free(cl);
```

Comment:
`errno` should be saved before this since `changelist_free()` may alter it.

Reply:
How? I see no way for `changelist_free()` to touch `errno`? None of the `uu_avl_*()` functions appear to touch it (errno is not even referenced in `lib/libuutil/uu_avl.c`.  `free()` does not touch it and `zfs_close()` does not touch it since it just does frees and `nvlist_free()` does not touch it.

Reply:
Ah you're right, my mistake. uu_panic() in uu_avl_remove() could but we're hosed at that point anyway.